packages/code-studio/src/styleguide/MockIrisGridTreeModel.ts

```diff
@@ -65,11 +65,11 @@ class MockIrisGridTreeModel
     return this.isEditable;
   }
 
-  setValues(edits: EditOperation[]): Promise<void> {
+  setValues(edits: EditOperation[]): Promise<never> {
     throw new Error('Method not implemented.');
   }
 
-  delete(ranges: GridRange[]): Promise<void> {
+  delete(ranges: GridRange[]): Promise<never> {
```

Comment:
I think these can just be `never` and it should be fine? It would just prevent you from trying to do anything w/ these operations when using the mock I believe

Looks like we would also want to add a flag to TS (or eslint) to make this an error for unreachable code. When I tried it out just now as unreachable, I don't get an error in TS. Looks like TS may allow it by default, but eslint disabled the rule since it used to be the default tsc behavior was to error on unreachable code.

https://github.com/typescript-eslint/typescript-eslint/issues/1041

Reply:
I'm not sure I understand this comment

Reply:
It doesn't really matter in this specific case since it's a mock for the styleguide, but here's an example for future things.

```ts
await delete([]);
console.log('Unreachable code');
```

If it's `Promise<never>`, then this code doesn't warn you that the 2nd line is unreachable.

If it's just `never`, then VSCode will tell you the 2nd line is unreachable since the 1st function returns never. Never is a subtype of every type, so anything that would be expecting a promise wouldn't complain about types, only unreachable code after it I believe.

Unreachable code isn't currently flagged in TS at least for us b/c ESLint disabled it since TS used to error on compile w/ unreachable code. They since changed that behavior it seems so we should either enable it in tsconfig or enable it in eslint for TS so we avoid having unreachable code